### PR TITLE
network: add support to dualstack (including fix for environments before 4.12)

### PR DESCRIPTION
**Pull request by creydr**
This PR adds a fix to #1389 to include only one VIP in the install-config in dual-stack environments <=4.11 (f1e0f51a2f982bd1180f4ec22e0a732158f0b89a).

Addition 1: It adds a fix in `add_dnsmasq_multi_entry()` to append the api vips, so both api vips are added to the dns_masq config file (9b796976f70c1ab182af14b404c909a8ab9f991d)
Addition 2: It renames the `API_VIP` and `INGRESS_VIP` fields to `API_VIPs` and `INGRESS_VIPs` to be clear these fields can contain multiple VIPs (012377813200b2b62419bf19eb182e2603077ecc).

Putting this on-hold until I finished all the tests in the different environments (4.11: v4,v6,v4v6, 4.12: v4,v6,v4v6)
/hold

### CODE REVIEW

**Comment by creydr** [comment]
# Testlog:

## OCP 4.11
### IPv4
```
$ cat config_chris.sh| egrep "IP_STACK=|OPENSHIFT_RELEASE_STREAM"
export OPENSHIFT_RELEASE_STREAM=4.11
export IP_STACK=v4

$ cat /etc/NetworkManager/dnsmasq.d/openshift-ostest.conf | egrep "^address"
address=/api.ostest.test.metalkube.org/192.168.111.5
address=/.apps.ostest.test.metalkube.org/192.168.111.4

$ oc -n kube-system get cm cluster-config-v1 -o yaml | egrep "apiVIP|ingressVIP"
        apiVIP: 192.168.111.5
        ingressVIP: 192.168.111.4
```

### IPv6
_pending_

### Dualstack
```
$ cat config_chris.sh| egrep "IP_STACK=|OPENSHIFT_RELEASE_STREAM"
export OPENSHIFT_RELEASE_STREAM=4.11
export IP_STACK=v4v6

$ oc -n kube-system get cm cluster-config-v1 -o yaml | egrep "apiVIP|ingressVIP"
        apiVIP: 192.168.111.5
        ingressVIP: 192.168.111.4

$ cat /etc/NetworkManager/dnsmasq.d/openshift-ostest.conf | egrep "^address"
address=/api.ostest.test.metalkube.org/192.168.111.5
address=/api.ostest.test.metalkube.org/fd2e:6f44:5dd8:c956::5
address=/.apps.ostest.test.metalkube.org/192.168.111.4
address=/.apps.ostest.test.metalkube.org/fd2e:6f44:5dd8:c956::4
```

## OCP 4.12
### IPv4
_pending_

### IPv6
_pending_

### Dualstack
_pending_

**Comment by cybertron** [comment]
Note that v4, v6, and v4v6 in 4.12 is covered by ci, so we shouldn't need to test those locally. Well, maybe v4v6 just to verify that we get two VIPs, but functionally those are covered.

One nit: It's unusual in this project to make a variable plural by adding a lower-case "s". Our other plural variables are all-caps. I wouldn't block it for that, but if we do need other changes to this PR we might think about making it consistent.

**Comment by cybertron** [comment]
I verified dualstack in 4.12 and confirmed that it still works in 4.11 too, so I think we're good to go. It's unlikely that this would have broken one single stack version and not the other.

/lgtm

File: network.sh [hunk 1]
```diff
@@ -193,31 +196,94 @@ if [ -n "${NETWORK_CONFIG_FOLDER:-}" ]; then
...
+    #     None
+    #
+    # Description:
+    #     Gets the INGRESS VIP and API VIP addresses (ipv4 and ipv6)
+    #
+    # Returns:
+    #     None
+    #
+    if [[ -n "${EXTERNAL_SUBNET_V4}" ]]; then
+        API_VIPs=$(dig +noall +answer "api.${CLUSTER_DOMAIN}" @$(network_ip ${BAREMETAL_NETWORK_NAME}) | awk '{print $NF}')
```
Comment (dtantsur) [inline network.sh:228]: nit: I don't really like using lower-case letters as part of an upper-case variable name

File: ocp_install_env.sh [hunk 2]
```diff
@@ -122,6 +122,58 @@ EOF
   fi
 }
 
+function renderVIPs() {
```
Comment (dtantsur) [inline ocp_install_env.sh:125]: very nit: I have a strong feeling we should rewrite the whole thing is Python

File: ocp_install_env.sh [hunk 3]
```diff
@@ -122,6 +122,58 @@ EOF
...
+    #     Second argument: value for the field
+    #
+    # Description:
+    #     This function helps to write in a YAML format multiple resources. (i.e: apiVIPs and ingressVIPs)
+    #     Example: apiVIPs, "192.168.11.5, fd2e:6f44:5dd8:c956::15"
+    #
+    # Returns:
+    #     YAML formatted resource
+    STRINGS_SEPARATOR=","
+    resource="${2}";
```
Comment (dtantsur) [inline ocp_install_env.sh:137]: nit: inconsistent case

**Pull request merged**